alibi_detect/base.py

```diff
@@ -264,3 +264,13 @@ def default(self, obj):
         elif isinstance(obj, (np.ndarray,)):
             return obj.tolist()
         return json.JSONEncoder.default(self, obj)
+
+
+class NotFitException(Exception):
+    """Exception raised when a transform is not fitted."""
+    pass
+
+
+class ThresholdNotInferredException(Exception):
+    """Exception raised when a threshold not inferred for an outlier detector."""
+    pass
```

Comment:
Would recommend creating a base exception class and then inheriting from that as it would allow app developers to selectively catch `AlibiDetectException`s only, e.g. see https://github.com/SeldonIO/alibi/blob/master/alibi/exceptions.py

Reply:
Oh and we should probably have naming consistency between the two libraries, e.g. in `alibi` I've opted for all library specific exceptions to be called `Error`, e.g. `NotFittedError`.